### PR TITLE
[Feature][Core] Support config encryption

### LINKED ISSUE #4124: The password of the configuration file needs ciphertext storage
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Encrypt the password in the configuration file to avoid storing the plaintext password in the file of the production environment. This is an effective security scheme, with necessary functions.

### Usage Scenario

Production release
https://github.com/apache/incubator-seatunnel/issues/3591
### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by TyrantLucifer**
## Purpose of this pull request

close #4124 

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by TyrantLucifer** [comment]
@EricJoy2048 @Hisoka-X @hailin0 PTAL

File: docs/en/connector-v2/Config-Encryption-Decryption.md [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+   ```shell
+   ${SEATUNNEL_HOME}/bin/seatunnel.sh --config config/v2.batch.template --encrypt
+   ```
+
+   Then you can see the encrypted configuration file in the terminal:
+
+   ```log
+   2023-02-20 17:50:58,319 INFO  org.apache.seatunnel.core.starter.command.ConfEncryptCommand - Encrypt config: 
+   {
```
Comment (hailin0) [inline docs/en/connector-v2/Config-Encryption-Decryption.md:86]: Output original file format?

example
```json
env {
}

source {

}

transform {

}

sink {

}
```

File: docs/en/connector-v2/Config-Encryption-Decryption.md [hunk 2]
```diff
@@ -0,0 +1,189 @@
+# Config  File Encryption And Decryption
+
+## Introduction
+
+In most production environments, sensitive configuration items such as passwords are required to be encrypted and cannot be stored in plain text, SeaTunnel provides a convenient one-stop solution for this.
+
+## How to use
```
Comment (hailin0) [inline docs/en/connector-v2/Config-Encryption-Decryption.md:7]: List the key value pair of encrypt & decrypt

example:
```json
username
password

*username
*password

...
```

File: docs/en/connector-v2/Config-Encryption-Decryption.md [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+   # Unless required by applicable law or agreed to in writing, software
+   # distributed under the License is distributed on an "AS IS" BASIS,
+   # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   # See the License for the specific language governing permissions and
+   # limitations under the License.
+   #
+
+   env {
+     execution.parallelism = 1
+     shade.identifier = "base64"
```
Comment (hailin0) [inline docs/en/connector-v2/Config-Encryption-Decryption.md:40]:
```suggestion
     shade.identifier = "base64"
     shade.password = "xxxxxxxxx"
     shade.keys = ["key1", "key2", ...]
```

Allow user to customize field list and password？

Comment (hailin0) [reply docs/en/connector-v2/Config-Encryption-Decryption.md:40]: shade.keys default value is : ["username", "password"]

File: docs/en/connector-v2/Config-Encryption-Decryption.md (hunk 1 above)
Comment (TyrantLucifer) [reply docs/en/connector-v2/Config-Encryption-Decryption.md:86]: This is limited by the format interface provided by typesafe config, this requirement I had concerned but not support this format.

File: docs/en/connector-v2/Config-Encryption-Decryption.md (hunk 2 above)
Comment (TyrantLucifer) [reply docs/en/connector-v2/Config-Encryption-Decryption.md:7]: Done, PTAL

File: docs/en/connector-v2/Config-Encryption-Decryption.md (hunk 3 above)
Comment (TyrantLucifer) [reply docs/en/connector-v2/Config-Encryption-Decryption.md:40]: According to the design of this interface, extend keys the user want to shade is controlled in the implemention of ConfigShade. All logic of encryption should contained in interface not config files. According to our actual production needs, complex configuration files increase the learning cost for users and encryption logic is not well controlled. So IMO, as-is is a better solution to decouple with config file. What do you think about it?

File: docs/en/connector-v2/Config-Encryption-Decryption.md [hunk 4]
```diff
@@ -0,0 +1,180 @@
+# Config  File Encryption And Decryption
```
Comment (EricJoy2048) [inline docs/en/connector-v2/Config-Encryption-Decryption.md:1]:
```suggestion
# Config File Encryption And Decryption
```

**Review by EricJoy2048 — APPROVED** [review]
LGTM

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**